### ISSUE TITLE
Getting "AH01070: Error parsing script headers" from web (Apache) container on certain pages with Drupal 8

### ISSUE THREAD

**Issue by lmakarov** [docksal/docksal#1110]
Logging mostly as FYI so others can find it.

This has been recently mentioned on Gitter. I also ran into this issue myself today:

```
[Mon Aug 05 13:44:36.060195 2019] [proxy_fcgi:error] [pid 12:tid 140074858306280] [client 192.168.160.2:46478] Premature end of script headers: index.php
[Mon Aug 05 13:44:36.060257 2019] [proxy_fcgi:error] [pid 12:tid 140074858306280] [client 192.168.160.2:46478] AH01070: Error parsing script headers
[Mon Aug 05 13:44:36.060263 2019] [proxy_fcgi:error] [pid 12:tid 140074858306280] (22)Invalid argument: [client 192.168.160.2:46478] AH01075: Error dispatching request to : 
192.168.160.2 - emmis [05/Aug/2019:13:44:35 +0000] "GET /admin/content/media HTTP/1.1" 500 -
```

The problem is with Drupal sending enormous headers and hitting the default 8k limit in Apache:
http://httpd.apache.org/docs/2.4/mod/core.html#limitrequestfieldsize

Increasing `LimitRequestFieldSize` to `16380` (or more) may alleviate the issue, however, it looks like this has to be addressed in Drupal.

Upstream issue with a proposed patch:
https://www.drupal.org/project/drupal/issues/2844620#comment-13126836

**Comment by paulsheldrake** [comment]
Just a heads up that this can also happen with D7 and the `pantheon_advanced_page_cache` module.  The work around at this point is to disable the module

**Comment by acteg** [comment]
I added patch but not resolve my problem, how i can increase value of LimitRequestFieldSize with docksal ?

**Comment by achekulaev** [comment]
@acteg https://docs.docksal.io/service/web/settings/
http://httpd.apache.org/docs/2.4/mod/core.html#limitrequestfieldsize

**Comment by acteg** [comment]
Thanks for reply @achekulaev achekulaev,

I tried to create new file .docksal/etc/apache/httpd-vhost-overrides.conf and put config 
```
LimitRequestFieldSize 20000
```
after each change i run 
```
 fin p restart web
```
But don't work ? i tried to connect to web ( fin bash web) and check if value is copied correctly :
```
/usr/local/apache2/conf/custom/httpd-vhost-overrides.conf
```
The value is correct and settings is copied correctly.

How i can check if value if used by apache ?

I tried higher values and always the same probeme error 500.

Thanks

**Comment by achekulaev** [comment]
@acteg try also increasing `LimitRequestLine 20000`

**Comment by achekulaev** [comment]
@acteg I believe you need to `fin reset web` to re-create it.

**Comment by acteg** [comment]
@achekulaev 
Yes i confirme the same Eroor 500 with the same issue Premature end of script ( see logs)
```
web_1        | [Fri Oct 18 08:43:20.198756 2019] [proxy_fcgi:error] [pid 9:tid 140616639285992] [client 172.22.0.11:53604] Premature end of script headers: index.php, referer: http://domains/fr/admin/content
web_1        | [Fri Oct 18 08:43:20.198978 2019] [proxy_fcgi:error] [pid 9:tid 140616639285992] [client 172.22.0.11:53604] AH01070: Error parsing script headers, referer: http://domains/fr/admin/content
web_1        | [Fri Oct 18 08:43:20.199079 2019] [proxy_fcgi:error] [pid 9:tid 140616639285992] (22)Invalid argument: [client 172.22.0.11:53604] AH01075: Error dispatching request to : , referer: http://domains/fr/admin/content
```

I tried fin reset web => don't work
stop/start stack  => don't work

How i can get value  used by apache ? any command for this ?

Thanks

**Comment by achekulaev** [comment]
@acteg you have the same Google as me

**Comment by leopitt** [comment]
I have had the same error and I believe this is caused by https://www.drupal.org/project/drupal/issues/2954339. The fix at https://www.drupal.org/project/drupal/issues/2954339#comment-12609045 resolved the problem for me.

Adding httpd-vhost-overrides.conf to my Docksal config didn't help.

**Comment by lpeabody** [comment]
The above was tagged as a duplicate of https://www.drupal.org/project/drupal/issues/2844620. The latest patch in that issue supposedly fixes the issue, but whether or not this is something that should also be applied to a non-development environment should be interpreted on a case-by-case basis. It might be that just setting `http.response.debug_cacheability_header` to `false` is all that is needed if the problem isn't occurring in production.

**Comment by JPustkuchen** [comment]
Yet I can also confirm this bug was caused and solved by the refreenced DO issues.

**Comment by boinkster** [comment]
From https://www.drupal.org/project/drupal/issues/2844620#comment-13354686, this worked for me:

For others experiencing this issue we noticed our development.servives.yml file had the following configuration:

http.response.debug_cacheability_headers: true

Changing this to false sorted the issue.

**Comment by MKorostoff** [comment]
Disabling debug_cacheability_headers worked for me, though for me it was located at sites/default/services.local.yml. This took me a while to figure out because both services.local.yml and development.services.yml existed in my code base, but only one was used.

**Comment by adaddinsane** [comment]
In our case it wasn't Drupal's fault - we have an insanely complex system which can easily generate a thousand or more cache tags for a page. And we use Fastly, so even with debug_cacheability_headers set false, it's still sending upwards of 5K in cache tags to Fastly (then add everything else in the headers).

Haven't solved it for us yet but I thought I'd add this in here in case someone else uses Fastly and is confused.

**Comment by inversed** [comment]
In my case, it was a combination of two overly large headers when running Apache 2.4:

- x-drupal-cache-tags
- Surrogate-Key (coming from pantheon_advanced_page_cache)

I needed the Drupal patch in issues/2954339 the x-drupal-cache-tags header and also the Remove HTTP headers module to strip out the Surrogate-Key header to get unblocked.

**Comment by lmakarov** [comment]
Closing, as this has to be handled on the Drupal side.

**Comment by alexcracea** [comment]
One more update on the issue, my team discovered that it is an issue with Apache https://bz.apache.org/bugzilla/show_bug.cgi?id=64919, if there is a potential fix it eventually require an update to the docksal image
